src/awkward/_v2/record.py

```diff
@@ -193,3 +193,40 @@ def to_list(self, behavior=None):
 
     def deep_copy(self):
         return Record(self._array.deep_copy(), copy.deepcopy(self._at))
+
+    def recursively_apply(
+        self,
+        action,
+        depth_context=None,
+        lateral_context=None,
+        keep_parameters=True,
+        numpy_to_regular=True,
+        return_array=True,
+        function_name=None,
+    ):
+
+        if return_array:
+            return Record(
+                self._array.recursively_apply(
+                    action,
+                    depth_context,
+                    lateral_context,
+                    keep_parameters,
+                    numpy_to_regular,
+                    True,
+                    function_name,
+                ),
+                self._at,
+            )
+
+        self._array.recursively_apply(
+            action,
+            depth_context,
+            lateral_context,
+            keep_parameters,
+            numpy_to_regular,
+            False,
+            function_name,
+        )
+
+        return None
```

Comment:
This is minor stuff, but I just thought it would be a little clearer this way. You've added a test, so that will ensure that this is not a broken commit.

```suggestion
        out = self._array.recursively_apply(
            action,
            depth_context,
            lateral_context,
            keep_parameters,
            numpy_to_regular,
            return_array,
            function_name,
        )

        if return_array:
            return Record(out, self._at)
        else:
            return None
```

Reply:
Huh that was the kind of thing I usually look for in a PR... thanks for making the change!